compiler/quilt/nodes.py

```diff
@@ -139,6 +140,38 @@ def _data(self):
 
         return store.load_dataframe(hash_list)
 
+def _create_filter_func(filter_dict):
+    filter_name = filter_dict.pop('name', None)
+    if filter_name is not None and not isinstance(filter_name, string_types):
+        raise ValueError("Invalid 'name'")
+
+    filter_meta = filter_dict.pop('meta', None)
+    if filter_meta is not None and not isinstance(filter_meta, dict):
+        raise ValueError("Invalid 'meta'")
+
+    if filter_dict:
+        raise ValueError("Unexpected data in the filter: %r" % filter_dict)
+
+    def helper(value, expected):
+        if isinstance(expected, dict):
+            if isinstance(value, dict):
+                for expected_key, expected_value in iteritems(expected):
+                    if not helper(value.get(expected_key), expected_value):
```

Comment:
recursive is most readable; wondering at what package depth we run out of call stack space? most packages should be pretty shallow so not worth doing this iteratively.

Reply:
Yeah... I'd wait till it becomes a problem.